Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+namespace ROS2
+{
+    void ROS2SpawnerComponent::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+
+        m_get_names_service = ros2Node->create_service<gazebo_msgs::srv::GetWorldProperties>(
+            "get_available_spawnable_names",
+            std::bind(&ROS2SpawnerComponent::GetAvailableSpawnableNames, this, AZStd::placeholders::_1, AZStd::placeholders::_2));
+
+        m_spawn_service = ros2Node->create_service<gazebo_msgs::srv::SpawnEntity>(
+            "spawn_entity", std::bind(&ROS2SpawnerComponent::SpawnRobot, this, AZStd::placeholders::_1, AZStd::placeholders::_2));
+    }
+
+    void ROS2SpawnerComponent::Deactivate()
+    {
+    }
+
+    void ROS2SpawnerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2SpawnerComponent, AZ::Component>()->Version(1)->Field("Spawner", &ROS2SpawnerComponent::m_spawnables);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2SpawnerComponent>("ROS2 Spawner", "Spawner component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "Allows spawning")
```

Comment:
extend description a bit - this should be human readable
"Facilitates spawning on robots in configurable locations" or something like this